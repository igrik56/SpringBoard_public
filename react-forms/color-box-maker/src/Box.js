import React from "react";

function Box ({id, width = 100, height = 100, handleRemove, backgroundColor = 'gray'}){
    const remove = () => handleRemove(id)

    return (
        <div className='box' style = {
            {
                width: `${width}px`,
                height: `${height}px`,
                backgroundColor
            }
        }>
            <div>
                <button onClick={remove}>X</button>
            </div>
        </div>
    )
}

export default Box