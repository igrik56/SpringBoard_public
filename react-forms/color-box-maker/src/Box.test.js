import React from "react";
import {render} from '@testing-library/react'
import Box from "./Box";

test('smoke test Box', () => {
    render(<Box />);
  });
  
  it('snapshot test Box', function(){
    const {asFragment} = render(<Box/>)
    expect (asFragment()).toMatchSnapshot()
  })