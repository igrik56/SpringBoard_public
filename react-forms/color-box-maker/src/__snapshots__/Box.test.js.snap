// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test Box 1`] = `
<DocumentFragment>
  <div
    class="box"
    style="width: 100px; height: 100px; background-color: gray;"
  >
    <div>
      <button>
        X
      </button>
    </div>
  </div>
</DocumentFragment>
`;
